sdk/storage/azure-storage-common/inc/azure/storage/common/internal/storage_bearer_token_authentication_policy.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/http/policies/policy.hpp>
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  class StorageBearerTokenAuthenticationPolicy final
+      : public Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy {
+  public:
+    /**
+     * @brief Construct a Storage Bearer Token challenge authentication policy.
+     *
+     * @param credential An #Azure::Core::TokenCredential to use with this policy.
+     * @param tokenRequestContext A context to get the token in.
+     * @param enableTenantDiscovery Enables tenant discovery through the authorization challenge.
+     */
+    explicit StorageBearerTokenAuthenticationPolicy(
+        std::shared_ptr<Azure::Core::Credentials::TokenCredential const> credential,
```

Comment:
```suggestion
        std::shared_ptr<const Azure::Core::Credentials::TokenCredential> credential,
```

Reply:
keep consistent with existing code

Reply:
For example, which existing code?
I keeped consistent with [BearerTokenAuthenticationPolicy](https://github.com/Azure/azure-sdk-for-cpp/blob/b33e8a9fd43a39762ac4e648336cd110c5fc58ab/sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp#L548) in current code.

Reply:
Fixed